app/controllers/test_with_your_users/submit_controller.js

```diff
@@ -44,6 +44,8 @@ module.exports = (req, res) => {
   const currencyMatch = AMOUNT_FORMAT.exec(paymentAmount)
```

Comment:
same. The fact that we had to repeat this function 3 times, makes me think we could actually extract it

Reply:
Ok I started doing this in this PR but actually I am going to it after I merge, as I have another PR that could benefit by being rebased on top of this